contracts/VaultOUSDLvUSD.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+/*
+  Vault is just a simple in/out it provide a fixed (non upgradable) address to transfer OUSD and lvUSD to (and pay interest into).
+  That means we need to change the architecture and add Minter or have the coordinator deal with than
+*/
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "hardhat/console.sol";
+
+contract VaultOUSDLvUSD {
+    address private oUSDContractAddress;
+    address private lvUSDContractAddress;
+
+    uint256 private oUSDRebasedBalance;
+    uint256 private lvUSDBalance;
+
+    constructor(address oUSDContract, address lvUSDContract) {
+        oUSDContractAddress = oUSDContract;
+        lvUSDContractAddress = lvUSDContract;
+        oUSDRebasedBalance = 0;
+        lvUSDBalance = 0;
+    }
+
+    modifier nonZeroAmount(uint256 amount) {
+        require(amount > 0, "Amount must be greater than zero");
+        _;
+    }
+
+    /* Privileged functions: Executive  */
+
+    /// @dev Deposits OUSD collateral into the vault.
+    ///
+    /// This function is only for the initial principle Deposits. It adds amount to oUSDRebasedBalance
+    /// Partners use ERC20 transfer to send OUSD as interest
+    ///
+    /// @param from address from which we transfer OUSD to vault .
+    /// @param amount the amount of collateral to deposit.
+    function depositOUSD(address from, uint256 amount)
+        external
+        nonZeroAmount(amount)
+    {
+        // Transfer funds from OUSD to Vault contract address
+        IERC20(oUSDContractAddress).transferFrom(from, address(this), amount);
+        addRebasedOUSDBalance(amount);
+    }
+
+    /// @dev Withdraw OUSD from vault and transfer it to an address.
+    ///
+    /// This function reverts if a deposit into the CDP was made in the same block. This is to prevent flash loan attacks
```

Comment:
Where is the logic for this revert? Also doesn't this limit our transaction volume assuming blocks take a while to form? Also wondering how this could affect a liquidity crisis scenario where lots of people are trying to exit but can't because only 1 per block... or is the "block" not what I am thinking... maybe I don't understand this.